pkg/policy/l4.go

```diff
@@ -733,19 +746,27 @@ func createL4Filter(policyCtx PolicyContext, peerEndpoints api.EndpointSelectorS
 
 	portName := ""
 	p := uint64(0)
+	endPort := p
 	if iana.IsSvcName(port.Port) {
 		portName = port.Port
 	} else {
 		// already validated via PortRule.Validate()
-		p, _ = strconv.ParseUint(port.Port, 0, 16)
+		ports := strings.Split(port.Port, "-")
+		if len(ports) >= 1 {
+			p, _ = strconv.ParseUint(ports[0], 0, 16)
+		}
+		if len(ports) == 2 {
+			endPort, _ = strconv.ParseUint(ports[1], 0, 16)
+		}
 	}
 
 	// already validated via L4Proto.Validate(), never "ANY"
 	u8p, _ := u8proto.ParseProtocol(string(protocol))
 
 	l4 := &L4Filter{
-		Port:                int(p),   // 0 for L3-only rules and named ports
-		PortName:            portName, // non-"" for named ports
+		Port:                uint16(p),       // 0 for L3-only rules and named ports
+		EndPort:             uint16(endPort), // 0 for a single port, >= 'Port' for a range
```

Comment:
Do we have a performance impact if `>= 'Port'`? Why didn't we make a requirement of having Endport **>** (and not **>=**) than Port

Reply:
No, there's no performance impact. The logic correctly identifies all cases that must be single selection (i.e. a full mask).